{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e525ea12",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty DataFrame\n",
      "Columns: [chr, start_pos, end_pos, mod, score, strand]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [chr, start_pos, end_pos, mod, score, strand]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [chr, start_pos, end_pos, mod, score, strand]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [chr, start_pos, end_pos, mod, score, strand]\n",
      "Index: []\n",
      "Empty DataFrame\n",
      "Columns: [chr, start_pos, end_pos, mod, score, strand]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "data = {\n",
    "    \"chr\" : [],\n",
    "    \"start_pos\" : [],\n",
    "    \"end_pos\" : [],\n",
    "    \"mod\" : [],\n",
    "    \"score\" : [],\n",
    "    \"strand\" : []\n",
    "}\n",
    "\n",
    "df_can = pd.DataFrame(data) #for chunks extraction from the canonical reads\n",
    "df_un_mod = pd.DataFrame(data) #for chunks extraction from the modified reads of non-modified G\n",
    "df_mod = pd.DataFrame(data) #for chunks extraction from the modified reads of the modified G's that are located in the middle\n",
    "df_can_A = pd.DataFrame(data) #for chunks extraction from the canonical reads of A's\n",
    "df_un_mod_A = pd.DataFrame(data) #for chunks extraction from the modified reads of A's\n",
    "\n",
    "#the reference file of the canonical reads\n",
    "fasta = open(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\ref_can.fa\", \"rt\")\n",
    "\n",
    "seq = \"\"\n",
    "\n",
    "p = 0 #can index\n",
    "q = 0 #can A index\n",
    "\n",
    "for line in fasta:\n",
    "    if line[0] == \">\":\n",
    "        seq = line[1:-1]\n",
    "        print(seq)\n",
    "    else:\n",
    "        for i in range(0,len(line)):\n",
    "            if line[i] == \"G\":\n",
    "                df_can.loc[p] = [seq, i, i+1, \"pos\", \"0\", \"+\"]\n",
    "                p=p+1\n",
    "            \n",
    "            if line[i] == \"A\":\n",
    "                df_can_A.loc[q] = [seq, i, i+1, \"pos\", \"0\", \"+\"]\n",
    "                q=q+1\n",
    "\n",
    "fasta.close()\n",
    "\n",
    "#choosing the G's that are really within the random area, 4 nucleotides inside - meaning that these G's also have random context.\n",
    "sub_can = df_can[df_can[\"start_pos\"] > 12]\n",
    "sub_can = df_can[df_can[\"start_pos\"] < 65]\n",
    "sub_can_A = df_can_A[df_can_A[\"start_pos\"] > 12]\n",
    "sub_can_A = df_can_A[df_can_A[\"start_pos\"] < 65]\n",
    "\n",
    "\n",
    "\n",
    "#the reference file of the modified reads\n",
    "fasta = open(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\ref_G_cA.fa\", \"rt\")\n",
    "\n",
    "seq = \"\"\n",
    "\n",
    "p = 0 #mod index\n",
    "q = 0 #un-mod-G index\n",
    "r = 0 #un-mod-A index\n",
    "\n",
    "for line in fasta:\n",
    "    if line[0] == \">\":\n",
    "        seq = line[1:-1]\n",
    "        print(seq)\n",
    "    else:\n",
    "        for i in range(0,len(line)):\n",
    "            if line[i] == \"G\":\n",
    "                if len(line) == 21: #the length of the reads from the can files\n",
    "                    df_mod.loc[p] = [seq, i, i+1, \"pos\", \"0\", \"+\"]\n",
    "                    p=p+1\n",
    "                else:\n",
    "                    df_un_mod.loc[q] = [seq, i, i+1, \"pos\", \"0\", \"+\"]\n",
    "                    q=q+1\n",
    "            \n",
    "            if line[i] == \"A\":\n",
    "                df_un_mod_A.loc[r] = [seq, i, i+1, \"pos\", \"0\", \"+\"]\n",
    "                r=r+1\n",
    "\n",
    "fasta.close()\n",
    "\n",
    "sub_un_mod = df_un_mod[df_un_mod[\"start_pos\"] > 12]\n",
    "sub_un_mod = df_un_mod[df_un_mod[\"start_pos\"] < 29]\n",
    "\n",
    "sub_mod = df_mod[df_mod[\"start_pos\"] > 12]\n",
    "sub_mod = df_mod[df_mod[\"start_pos\"] < 29]\n",
    "\n",
    "sub_un_mod_A = df_un_mod_A[df_un_mod_A[\"start_pos\"] > 12]\n",
    "sub_un_mod_A = df_un_mod_A[df_un_mod_A[\"start_pos\"] < 29]\n",
    "\n",
    "\n",
    "sub_can.to_csv(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\focus_can_G.bed\", sep='\\t', index=False, header=False)\n",
    "sub_un_mod.to_csv(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\focus_un_mod_G_cA.bed\", sep='\\t', index=False, header=False)\n",
    "sub_mod.to_csv(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\focus_mod_cA.bed\", sep='\\t', index=False, header=False)\n",
    "sub_can_A.to_csv(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\focus_can_A.bed\", sep='\\t', index=False, header=False)\n",
    "sub_un_mod_A.to_csv(\"C:\\\\Users\\\\NirLavi\\\\Desktop\\\\PhD\\\\Lab\\\\IL LAB\\\\O6Meg\\\\data\\\\exp2_randomers\\\\alternative-training\\\\focus_un_mod_A_cA.bed\", sep='\\t', index=False, header=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
